appinventor/appengine/src/com/google/appinventor/client/TopPanel.java

```diff
@@ -99,126 +107,50 @@ public TopPanel() {
      *  |+----------++---------------++-----------------+|
      *  +------------------------------------------------+
      */
-    HorizontalPanel topPanel = new HorizontalPanel();
-    topPanel.setVerticalAlignment(HorizontalPanel.ALIGN_MIDDLE);
-
-    // Create the Tools
-    TopToolbar tools = new TopToolbar();
-    ode.setTopToolbar(tools);
-
-    // Create the Links
-    HorizontalPanel links = new HorizontalPanel();
-    links.setStyleName("ode-TopPanelLinks");
-    links.setVerticalAlignment(HorizontalPanel.ALIGN_MIDDLE);
-
-    if (Ode.getInstance().isReadOnly()) {
-      Label readOnly = new Label(MESSAGES.readOnlyMode());
-      readOnly.setStyleName("ode-TopPanelWarningLabel");
-      links.add(readOnly);
-    }
-
-    // My Projects Link
-    TextButton myProjects = new TextButton(MESSAGES.myProjectsTabName());
-    myProjects.setStyleName("ode-TopPanelButton");
+    Config config = getSystemConfig();
+    initWidget(UI_BINDER.createAndBindUi(this));
```

Comment:
Since config will be used on the following line, consider swapping the order of these two lines since they can be performed independently.